### PR TITLE
Add version information to server and CLI

### PR DESCRIPTION
**Pull request by stefannica**
* version information is injected at build time both in server and CLI
* add `/version` API path
* add CLI `fuseml version` command to print the version of both client
and server (if available):

```
> bin/fuseml version -u http://localhost:8000
---
client:
  version: unknown
  gitCommit: efd3e60d
  buildDate: 2021-06-02T11:35:07Z
  goVersion: go1.16
  compiler: gc
  platform: linux/amd64
server:
  version: unknown
  gitcommit: efd3e60d
  builddate: 2021-06-02T11:35:07Z
  golangversion: go1.16
  golangcompiler: gc
  platform: linux/amd64
```

### CODE REVIEW

File: pkg/version/version.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	Version string = "unknown"
+
+	// GitCommit is the git commit ID
+	GitCommit string = "unknown"
+
+	// BuildDate is the date when the binary was built
+	BuildDate string = "unknown"
+)
+
+// VersionInfo contains versioning information.
```
Comment (jsuchome) [inline pkg/version/version.go:20]: Could you fix the lint error?

File: Makefile [hunk 2]
```diff
@@ -11,7 +11,28 @@ endif
...
+GIT_TAG    = $(shell git describe --tags --abbrev=0 --exact-match 2>/dev/null)
+BUILD_DATE = $(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
+
+LDFLAGS += -X $(PKG_PATH)/version.GitCommit=$(GIT_COMMIT)
+LDFLAGS += -X $(PKG_PATH)/version.BuildDate=$(BUILD_DATE)
+
+ifdef VERSION
+	BINARY_VERSION = $(VERSION)
+endif
+BINARY_VERSION ?= ${GIT_TAG}
```
Comment (flaviodsr) [inline Makefile:28]: can you make it that when building from the `main` branch set the version to `dev`? just to keep it in sync with the `dev` tag used for the container image.

File: pkg/version/version.go (hunk 1 above)
Comment (stefannica) [reply pkg/version/version.go]: done

**Review by flaviodsr — APPROVED** [review]
LGTM, just a suggestion.

**Review by jsuchome — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (stefannica) [reply Makefile]: ok

**Review by flaviodsr — DISMISSED** [review]
Very nice 👏

**Review by jsuchome — APPROVED** [review]

**Review by flaviodsr — APPROVED** [review]

**Pull request merged**